File: ejercicio03/gretting/src/Greeting2.java
/*
Tecnológico Nacional de México
Instituto Tecnológico de León, Campus 1
Carrera: Ingeniería en Sistemas Computacionales
Materia: Topicos Avanzados de Programacion
Horario:
    Martes y Jueves de 7:00 a 8:45
    Viernes de 7:50 a 8:40
Alumno: Marlene Inés Moreno Velázquez
Ejercicio #3
Fecha: 25/08/2026
*/

import java.awt.FlowLayout;

import javax.swing.JFrame;
import javax.swing.JLabel;
import javax.swing.JTextField;

public class Greeting2 extends JFrame {
  JTextField textField;
  JLabel label;

  void init() {   // define variables del programa y despliegue de meta
    String msg = "Para mostrar un campo de texto en una ventana simple";

    label = new JLabel(msg);
  }

  public static void main (String[] args) {
    Greeting2 frame = new Greeting2();

    frame.init();
    frame.createGUI();
    frame.addWidgets();
    frame.setVisible(true); 
  }

  void addWidgets() {    // despliega la vista
    textField = new JTextField("Hello! primera GUI");
    add(label);
    add(textField);   
  }

  void createGUI() {  // define contexto grafico de la vista
    setSize(300, 200);           
    setDefaultCloseOperation(EXIT_ON_CLOSE);
    setLayout(new FlowLayout() );
  }
}

File: ejercicio04/Greeting2.py
import tkinter as tk


class Greeting2(tk.Tk):

  def __init__(self):
    super().__init__()
    self.textField = None
    self.label = None

  def init(self):  # define variables del programa y despliegue de meta
    msg = "Para mostrar un campo de texto en una ventana simple"
    self.label = tk.Label(self, text=msg)

  def createGUI(self):  # define contexto grafico de la vista
    self.geometry("300x200")

  def addWidgets(self):  # despliega la vista
    self.textField = tk.Entry(self)
    self.textField.insert(0, "Hello! primera GUI")


    self.label.pack()
    self.textField.pack()


if __name__ == "__main__":
  frame = Greeting2()

  frame.init()
  frame.createGUI()
  frame.addWidgets()
  frame.mainloop()  